src/integrationTest/java/org/opensearch/security/ThreadPoolTests.java

```diff
@@ -0,0 +1,88 @@
+import org.opensearch.security.http.ExampleSystemIndexPlugin;
+import org.opensearch.test.framework.TestSecurityConfig.AuthcDomain;
+import org.opensearch.test.framework.cluster.ClusterManager;
+import org.opensearch.test.framework.cluster.LocalCluster;
+import org.opensearch.test.framework.cluster.TestRestClient;
+import org.opensearch.test.framework.cluster.TestRestClient.HttpResponse;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.not;
+import static org.opensearch.security.support.ConfigConstants.SECURITY_RESTAPI_ROLES_ENABLED;
+import static org.opensearch.test.framework.TestSecurityConfig.Role.ALL_ACCESS;
+import static org.opensearch.test.framework.TestSecurityConfig.User.USER_ADMIN;
+
+@RunWith(com.carrotsearch.randomizedtesting.RandomizedRunner.class)
+@ThreadLeakScope(ThreadLeakScope.Scope.NONE)
+public class ThreadPoolTests {
+
+    public static final AuthcDomain AUTHC_DOMAIN = new AuthcDomain("basic", 0).httpAuthenticatorWithChallenge("basic").backend("internal");
+
+    @ClassRule
+    public static final LocalCluster cluster = new LocalCluster.Builder().clusterManager(ClusterManager.SINGLENODE)
+        .anonymousAuth(false)
+        .authc(AUTHC_DOMAIN)
+        .users(USER_ADMIN)
+        .plugin(ExampleSystemIndexPlugin.class)
+        .nodeSettings(Map.of(SECURITY_RESTAPI_ROLES_ENABLED, List.of("user_" + USER_ADMIN.getName() + "__" + ALL_ACCESS.getName())))
+        .build();
+
+    @Test
+    public void testEnsureNoThreadLeftRunningInGenericThreadPool() throws IOException, InterruptedException {
+        try (TestRestClient client = cluster.getRestClient(USER_ADMIN)) {
+            client.put("test-index");
+
+            XContentBuilder builder = XContentFactory.jsonBuilder();
+            builder.startObject();
+            builder.field("field1", "foo");
+            builder.endObject();
+
+            HttpResponse indexDocResponse = client.putJson("test-index/_doc/1", builder.toString());
+
+            assertThat(indexDocResponse.getStatusCode(), equalTo(RestStatus.CREATED.getStatus()));
+
+            XContentBuilder updateBuilder = XContentFactory.jsonBuilder();
+            updateBuilder.startObject();
+            updateBuilder.startObject("doc");
+            updateBuilder.field("field1", "bar");
+            updateBuilder.endObject();
+            updateBuilder.endObject();
+
+            HttpResponse updateDocResponse = client.postJson("test-index/_update/1", updateBuilder.toString());
+
+            assertThat(updateDocResponse.getStatusCode(), equalTo(RestStatus.OK.getStatus()));
+
+            client.delete("test-index");
+
+            Thread.sleep(2000);
+
+            HttpResponse hotThreadsResponse = client.get("_nodes/hot_threads");
+
+            assertThat(hotThreadsResponse.getBody(), not(containsString("ClusterStateMetadataDependentPrivileges")));
```

Comment:
The assertion here is very specific and I'd like to make it more generalizable, but I ran into issues running the `integrationTest` check on the windows runners. I would like to get this PR merged to resolve the issues seen in ISM CI checks and work to generalize these assertions in a follow-up PR.

Reply:
Yes, I was thinking similarly. I could imagine this with a unit test directly on `DlsFlsProcessedConfig` which verifies that the thread spawned by `updateClusterStateMetadataAsync` terminates. That should be possible with a mocked `ThreadPool`  instance. 

I wanted to try that already yesterday, but just had too many things on my desk. I will try to work it out next week.